hare/broker.go

```diff
@@ -77,49 +84,61 @@ func (broker *Broker) Start() error {
 	return nil
 }
 
-type Message struct {
-	msg            *pb.HareMessage
-	bytes          []byte
-	validationChan chan service.MessageValidation
-}
-
-func (msg Message) reportValidationResult(isValid bool) {
-	if msg.validationChan != nil {
-		msg.validationChan <- service.NewMessageValidation(msg.bytes, ProtoName, isValid)
-	}
-}
-
 // Dispatch incoming messages to the matching set id instance
 func (broker *Broker) dispatcher() {
 	for {
 		select {
 		case msg := <-broker.inbox:
+			if msg == nil {
+				log.Warning("Message validation failed: called with nil")
+				msg.ReportValidation(ProtoName, false)
+				continue
+			}
+
 			hareMsg := &pb.HareMessage{}
 			err := proto.Unmarshal(msg.Bytes(), hareMsg)
 			if err != nil {
 				log.Error("Could not unmarshal message: ", err)
```

Comment:
shouldn't you add ReportValidation here as well

Reply:
Yes. Fixed.